src/stepfunctions/steps/states.py

```diff
@@ -73,11 +73,33 @@ def to_dict(self):
                 k = to_pascalcase(k)
                 if k == to_pascalcase(Field.Parameters.value):
                     result[k] = self._replace_placeholders(v)
+                elif self._is_placeholder_compatible(k):
+                    if isinstance(v, Placeholder):
+                        result[k] = v.to_jsonpath()
+                    else:
+                        result[k] = v
```

Comment:
I think this would be clearer without nested if statements. The final `else` is the same as the one already on line 81
```suggestion
                elif self._is_placeholder_compatible(k) and isinstance(v, Placeholder):
                    result[k] = v.to_jsonpath()

```

Reply:
Indeed - this is clearer!